### PR TITLE
Onboarding: show the draft post modal when the user choose write intent

### PR DESCRIPTION
**Pull request by arthur791004**
#### Changes proposed in this Pull Request

* Implement `<DraftPostModal />` component with the following behaviors:
  * Click [X] to verify the modal is closed and stay in the editor
  * Click [Start writing] to verify the modal closed and stay in the editor
  * Click [I'm not ready] to verify you go to My Home
* When the user selects the write intent and  finishes the setup-site flow, it will set `wpcom_signup_complete_show_draft_post_modal` to session storage and redirect to the editor. When landing on the editor, it will read that value and cleanup the session storage. Next, using this value to set `showDraftPostModal` in query args, and then the `wpcom-block-editor-nux` can retrieve this value and decide to show the `<DraftPostModal />` or `<LaunchWpcomWelcomeTour />`. Besides, as the value is cleaned up immediately after reading, the `<DraftPostModal />` will only show at the first time. When the user lands on the editor next time, it will show the `<LaunchWpcomWelcomeTour />`.
* The reason I use `sessionStorage` is that I just want to keep the URL looks clean and we can easily cleanup the value without any redirecting.

https://user-images.githubusercontent.com/13596067/136190238-deac32e1-899f-4402-9df5-51db59800244.mov

image

image

#### Testing instructions

* Sandbox your API and ensure your siteSlug domain point to your sandbox
* Open another terminal and run `cd apps/editing-toolkit && yarn dev --sync`
* Go to `/start/setup-site?siteSlug=arthurtestonboarding.wordpress.com&flags=signup/hero-flow`
* Select write intent and you'll redirect to post editor
* Verify write your post modal show
* Verify the following behaviors one by one (note that when you finish one, you need to repeat to choose the intent)
  * Click [X] to verify the modal is closed and stay in the editor
  * Click [Start writing] to verify the modal closed and stay in the editor
  * Click [I'm not ready] to verify you go to My Home
* Refresh the page and verify the write your post modal not show but welcome tour shows

Related to https://github.com/Automattic/wp-calypso/issues/55603

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { useI18n } from '@wordpress/react-i18n';
+import draftPostImage from './images/draft-post.svg';
+import './style.scss';
+
+const DraftPostModal = () => {
+	const { __ } = useI18n();
+
+	return (
+		<Modal open className="wpcom-block-editor-draft-post-modal">
+			<img src={ draftPostImage } alt={ __( 'Draft post', 'full-site-editing' ) } />
```
Comment (a8ci18n) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'Draft posts.' )` *ES Score: 8*

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	return (
+		<Modal
+			open={ isOpen }
+			className="wpcom-block-editor-draft-post-modal"
+			onRequestClose={ closeModal }
+		>
+			<div className="wpcom-block-editor-draft-post-modal__image-container">
+				<img src={ draftPostImage } alt={ __( 'Draft post', 'full-site-editing' ) } />
```
Comment (a8ci18n) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'Draft posts.' )` *ES Score: 8*

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+			</div>
+			<h1 className="wpcom-block-editor-draft-post-modal__title">
+				{ __( 'Write your first post', 'full-site-editing' ) }
```
Comment (a8ci18n) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js:28]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( '3. Write your first post' )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/feat/onboarding-draft-post-modal)

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	return (
+		<Modal
+			open={ isOpen }
+			className="wpcom-block-editor-draft-post-modal"
+			onRequestClose={ closeModal }
+		>
+			<div className="wpcom-block-editor-draft-post-modal__image-container">
+				<img src={ draftPostImage } alt={ __( 'Write post', 'full-site-editing' ) } />
```
Comment (a8ci18n) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/index.js:25]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **44** times:
`translate( 'Write a Post' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/feat/onboarding-draft-post-modal)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6734751

Thank you @arthur791004 for including a screenshot in the description! This is really helpful for our translators.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/style.scss [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+@import '@wordpress/base-styles/breakpoints';
+@import '@wordpress/base-styles/mixins';
+
+.wpcom-block-editor-draft-post-modal {
+	.components-modal__header {
+		height: auto;
+		padding: 10px;
+		border-bottom: 0;
+
+		button {
```
Comment (roo2) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/style.scss:11]: I wonder if this should be applied to all modals in future, it seems to make the modal more usable in general 👍

**Review by roo2 — APPROVED** [review]
Tested, LGTM! I didn't like the session storage approach at first, but now realize it's the best option I can see for implementing this 👍.

I think missing in the test instructions was also syncing the ETK with `cd apps/editing-toolkit && yarn dev --sync`

**Comment by arthur791004** [comment]
> Open another terminal and run cd apps/wpcom-block-editor && yarn dev --sync

@roo2 I found I type the wrong folder 😅

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/style.scss (hunk 5 above)
Comment (arthur791004) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/draft-post-modal/style.scss:11]: I think we can consider this one when doing post published modal

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6734751).

**Pull request merged**